xdsl/transforms/loop_nest_lowering_utils.py

```diff
@@ -54,32 +49,140 @@ def indices_for_map(
+    `make_body` function.
+    If `iter_args` are passed in, the loop nest will pass them from parent loop to child
+    loop, and the results of `make_body` are expected to be equal in length to
+    `iter_args`.
+
+    `zero_op` and `one_op` are operations ƒor the `0` and `1` index constants for the
+    loop nest lower bound and step. The upper bounds are specified by the `bounds`
+    arguement.
+    """
+    if not bounds:
+        return make_body(rewriter, insertion_point, (), iter_args)
+
+    iter_arg_types = tuple(arg.type for arg in iter_args)
+    loops: list[scf.For] = []
+    index = IndexType()
+
+    for i, ub in enumerate(bounds):
+        loop = scf.For(
+            zero_op.result,
+            ub,
+            one_op.result,
+            iter_args,
+            Region(Block(arg_types=(index, *iter_arg_types))),
+        )
+        iter_args = loop.body.block.args[1:]
+        loops.append(loop)
+        rewriter.insert_op_at_location(loop, insertion_point)
+        results = loop.results
+
+        if i + 1 == len(bounds):
+            # Innermost loop iteration
+            results = make_body(
+                rewriter,
+                InsertPoint.at_start(loop.body.block),
+                tuple(loop.body.block.args[0] for loop in loops),
+                iter_args,
+            )
+            if len(results) != len(iter_args):
+                raise ValueError(
+                    "Unexpected number of results from `make_body` helper "
+                    f"({len(results)}), expected {len(iter_args)}"
+                )
+        rewriter.insert_op_at_end(scf.Yield(*results), loop.body.block)
+        insertion_point = InsertPoint.at_start(loop.body.block)
+
+    return loops[0].results
+
+
+def _insert_load_ops(
+    rewriter: PatternRewriter,
+    insertion_point: InsertPoint,
+    ind_vars: Sequence[BlockArgument],
+    affine_map_attrs: Sequence[AffineMapAttr],
+    operands: Sequence[SSAValue],
+    args: Sequence[BlockArgument],
+    load: Callable[
+        [SSAValue, Sequence[SSAValue], PatternRewriter, InsertPoint], SSAValue
+    ],
+) -> Sequence[tuple[int, SSAValue]]:
+    res: list[tuple[int, SSAValue]] = []
+    for i, (affine_map_attr, operand, arg) in enumerate(
+        zip(affine_map_attrs, operands, args, strict=True)
+    ):
+        if not arg.uses:
+            continue
+        affine_map = affine_map_attr.data
+        indices = indices_for_map(rewriter, insertion_point, affine_map, ind_vars)
+        res_val = load(operand, indices, rewriter, insertion_point)
+        res.append((i, res_val))
+    return res
+
+
+def _insert_store_ops(
+    rewriter: PatternRewriter,
+    insertion_point: InsertPoint,
+    ind_vars: Sequence[BlockArgument],
+    output_indexing_maps: Sequence[AffineMapAttr],
+    yield_operands: Sequence[SSAValue],
+    output_operands: Sequence[SSAValue],
+    store: Callable[[SSAValue, SSAValue, Sequence[SSAValue]], Operation],
+):
```

Comment:
Same here, seeing this many operands without a docstring makes me nervous

Reply:
Why does it need all these things to insert the stores (e.g. what are the `yield_operands`?)